src/ipbb/cmds/repo.py

```diff
@@ -285,6 +285,14 @@ def git(env, repo, branch, dest):
         echo('Checking out branch/tag ' + style(branch, fg='blue'))
         sh.git('checkout', branch, '-q', _out=sys.stdout, _cwd=lRepoLocalPath)
 
+    elif revision is not None:
+        echo('Checking out revision ' + style(revision, fg='blue'))
+        try:
+            sh.git('checkout', revision, '-q', _out=sys.stdout, _cwd=lRepoLocalPath)
+        except Exception as err:
+            secho("Failed to check out requested revision. Staying on master.", fg='red')
+            sh.git('checkout', 'master', '-q', _out=sys.stdout, _cwd=lRepoLocalPath)
```

Comment:
Is this additional checkout required? Shouldn't the call to `git clone` have already checked out the default branch?

Reply:
In principle yes. As long as one trusts that the failed checkout did not alter the state of the clone in any way. I'm not sure if this is guaranteed, so I explicitly reverted to the master branch.

Reply:
Ok. Now, defaulting to `master` is not guaranteed to be possible. The only way that I can see this working under all scenarios is to query the name of the current branch after clone, and if this checkout fail, re-checkout the cloned branch.
Does this make sense?

Reply:
See above.